ocs_ci/ocs/cluster.py

```diff
@@ -743,3 +743,92 @@ def validate_osd_utilization(osd_used=80):
             logger.warn(f"{osd} used value {value}")
 
     return _rc
+
+
+def get_pgs_per_osd():
+    """
+    Function to get ceph pg count per OSD
+
+    Returns:
+        osd_dict (dict): Dict of osd name and its used value
+        i.e {'osd.0': 136, 'osd.2': 136, 'osd.1': 136}
+
+    """
+    osd_dict = {}
+    ceph_cmd = "ceph osd df"
+    ct_pod = pod.get_ceph_tools_pod()
+    output = ct_pod.exec_ceph_cmd(ceph_cmd=ceph_cmd)
+    for osd in output.get('nodes'):
+        osd_dict[osd['name']] = osd['pgs']
+
+    return osd_dict
+
+
+def get_balancer_eval():
+    """
+    Function to get ceph pg balancer eval value
+
+    Returns:
+        eval_out (float): Eval output of pg balancer
+
+    """
+    ceph_cmd = "ceph balancer eval"
+    ct_pod = pod.get_ceph_tools_pod()
+    eval_out = ct_pod.exec_ceph_cmd(ceph_cmd=ceph_cmd).split(' ')
+    return float(eval_out[3])
+
+
+def get_pg_balancer_status():
+    """
+    Function to check pg_balancer active and mode is upmap
+
+    Returns:
+        bool: True if active and upmap is set else False
+
+    """
+    # Check either PG balancer is active or not
+    ceph_cmd = "ceph balancer status"
+    ct_pod = pod.get_ceph_tools_pod()
+    output = ct_pod.exec_ceph_cmd(ceph_cmd=ceph_cmd)
+
+    # Check 'mode' is 'upmap', based on suggestion from Ceph QE
+    # TODO: Revisit this if mode needs change.
+    if output['active'] and output['mode'] == 'upmap':
+        logging.info("PG balancer is active and mode is upmap")
+        return True
+    else:
+        logging.error("PG balancer is not active")
+        return False
+
+
+def validate_pg_balancer():
+    """
+    Validate either data is equally distributed to OSDs
+
+    Returns:
+        bool: True if osd data consumption difference is <= 2% else False
```

Comment:
Nice!, we can also take the percent as param if that helps later on, but for now looks good.